### ISSUE TITLE
What happens if we manually regenerate the ACR tokens?

### ISSUE THREAD

**Issue by htwashere** [tc-ca/TC3oE-Kubernetes#2]
Our AKS is not deploying the containers/pod properly. In most case, we get the "ImagePullBackOff" - trying and failing to pull image from acr repository. I am assuming that we are using the token/ScopeMap feature so we tried to manually re-generate the ACR token. I can use this new token and manually do a docker login , do Docker pull successfully. However, if we try the deployment again in AKS, we are still getting the same error.

Is there a place where we need to maybe sync the token into the framework?

Also, I am curious: how does AKS (or ArgoCD?) know which login/token to use when it is deploying? I do not seem to find properties in the various yaml file to mention the token or login id? or, is it using RBAC acr-pull somewhere instead? 

Thanks.

**Comment by TeamDman** [comment]
ACR scope maps and tokens are for the service connections for _pushing_ images in Azure devops pipelines. If you're getting imagepullbackoff, make sure the "ACR pull" role assignment exists on the ACR (should be created automatically by Terraform, granting the role to the Kubelet identity of the cluster"
The role assignment probably exists since Terraform would complain if it had trouble making it, so it's probably a different problem - make sure the image being referenced by the pod manifest is present in the ACR with no typos in tepository, image name, or image tag. There might be logs with additional details in a lube-system pod or something

**Comment by htwashere** [comment]
Yes, above makes sense based on your design. We originally tried to debugging by checking the "ACR pull" role assignment. I noticed there are different user assigned MI being available, and yes, it was assigned properly by terraform. By playing with that, it did not solve the problem, hence I started going down the scope maps and token route. I will continue to troubleshoot and report back once I find the issue. Thank you!
